### PR TITLE
Fix issue with proxy hanging

### PR DESCRIPTION
**Pull request by cjellick**
If a connection was opened but no data was sent, the proxy_protocol
logic would hang forever waiting for data. Because this logic was
called by RemoteAddr(), which is called in the main loop of
Server.Server(), it caused the entire proxy to hang.

Addresses: https://github.com/rancher/rancher/issues/1836

### CODE REVIEW

File: proxy/integration_test.go [hunk 1]
```diff
@@ -59,6 +61,7 @@ func TestMain(m *testing.M) {
 
 	router := mux.NewRouter()
 	router.HandleFunc("/v1/subscribe", getWsHandler())
+	router.HandleFunc("/v1/proxyproto", proxyProtoHandler)
```
Comment (ibuildthecloud) [inline proxy/integration_test.go:64]: for testing?

Comment (ibuildthecloud) [inline proxy/integration_test.go:64]: oh duh, it's in the file integration_test

File: proxy/proxyprotocol/proxy_protocol.go [hunk 2]
```diff
@@ -186,46 +167,43 @@ func (p *Conn) checkPrefix() error {
 	}
 
 	// Verify the type is known
-	switch parts[1] {
-	case "TCP4":
-	case "TCP6":
-	default:
+	if parts[1] != "TCP4" && parts[1] != "TCP6" {
```
Comment (ibuildthecloud) [inline proxy/proxyprotocol/proxy_protocol.go:170]: lol

File: proxy/proxyprotocol/http_integration.go [hunk 3]
```diff
@@ -4,23 +4,39 @@ import (
...
+
+const (
+	xForwardedProto string = "X-Forwarded-Proto"
+	xForwardedPort  string = "X-Forwarded-Port"
+	xForwardedFor   string = "X-Forwarded-For"
+	sep             string = ", "
 )
 
 func AddHeaders(req *http.Request, httpsPorts map[int]bool) {
 	proxyProtoInfo := getInfo(req.RemoteAddr)
```
Comment (ibuildthecloud) [inline proxy/proxyprotocol/http_integration.go:18]: RemoteAddr is the LB address, right?  So this isn't unique anymore?

**Comment by cjellick** [comment]
Begrudging offline LGTM from @ibuildthecloud

**Pull request merged**